extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergWriteInstructions.java

```diff
@@ -42,11 +45,28 @@ public boolean createTableIfNotExist() {
     public abstract Optional<Boolean> verifySchema();
     // @formatter:on
 
+    /**
+     * A one-to-one {@link Map map} from Deephaven to Iceberg column names to use when writing deephaven tables to
+     * Iceberg tables.
+     */
+    public abstract Map<String, String> dhToIcebergColumnRenames();
```

Comment:
Ditto, this is also tied w/ a specific Schema. From a configuration point, I see the ease-of-use for using strings, but I wonder if we should have `Map<NestedField, String>`, or `Map<Integer, String> + Schema` (user can still use strings, but we materialize into this). I also suggest we model it with the iceberg data as the key, since it's the target system that is dictating uniqueness in this case (otherwise, we need to add a check that there are no duplicate values). Technically, we _could_ support writing a single DH column to multiple iceberg columns, although I don't see a reason to offer that out of the gate without a clear use case.

Reply:
I like this idea, although I would prefer to do this change as part of a separate PR for both reading and writing side together. For now, I have added a check for uniqueness for no duplicate values.
I can also link these comments in the issue #6124.